src/Functions/byteSwap.cpp

```diff
@@ -0,0 +1,79 @@
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionUnaryArithmetic.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+extern const int NOT_IMPLEMENTED;
+}
+
+namespace
+{
+template <typename T>
+requires std::is_same_v<T, UInt8>
```

Comment:
By the way, there is `std::byteswap` that works for all supported data types - let's use it to simplify.

Reply:
Isn't that only available from C++23 onwards?

Reply:
Yes, we use C++23.

Reply:
Oh, awesome! This might be a stupid question but how exactly does one figure out which version of C++ a project is compatible with at any given point (I'm quite new to C++ development)?

Reply:
I'll use `std::byteswap`. FYI, it even accepts signed integers. This means that people will be able to use byteswap with negative integers as well. Example:
```
┌─byteSwap(-255)─┐
│            511 │
└────────────────┘
```

Reply:
Ah, makes sense. Thanks!